### ISSUE TITLE
Unable to use logfiles in ShellCommand after upgraded buildbot worker to python 3

### ISSUE THREAD

**Issue by doberkofler** [buildbot/buildbot#4836]
I have just upgraded a buildbot worker to Python version 3.7.3 and buildbot version 2019.06.13 on the Windows platform and without any changes in the configuration itself, I now get the following buildbot error whenever the `logfiles` parameter is used in a `ShellCommand`

```
2019-06-14 11:30:12+0200 [-] Unhandled Error
	Traceback (most recent call last):
	  File "c:\program files\python37\lib\site-packages\twisted\internet\base.py", line 1272, in run
	    self.mainLoop()
	  File "c:\program files\python37\lib\site-packages\twisted\internet\base.py", line 1281, in mainLoop
	    self.runUntilCurrent()
	  File "c:\program files\python37\lib\site-packages\twisted\internet\base.py", line 902, in runUntilCurrent
	    call.func(*call.args, **call.kw)
	  File "c:\program files\python37\lib\site-packages\twisted\internet\task.py", line 239, in __call__
	    d = defer.maybeDeferred(self.f, *self.a, **self.kw)
	--- <exception caught here> ---
	  File "c:\program files\python37\lib\site-packages\twisted\internet\defer.py", line 151, in maybeDeferred
	    result = f(*args, **kw)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 166, in poll
	    self.command.addLogfile(self.name, data)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 773, in addLogfile
	    self._addToBuffers(('log', name), data)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 749, in _addToBuffers
	    self._sendBuffers()
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 731, in _sendBuffers
	    self._sendMessage(msg)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 677, in _sendMessage
	    msg = self._collapseMsg(msg)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\runprocess.py", line 663, in _collapseMsg
	    m = bytes2unicode(m, self.builder.unicode_encoding)
	  File "c:\program files\python37\lib\site-packages\buildbot_worker\compat.py", line 87, in bytes2unicode
	    return text_type(x, encoding, errors)
	builtins.UnicodeDecodeError: 'utf-8' codec can't decode byte 0xfc in position 3388: invalid start byte
```

Example step:
```
	factory.addStep(steps.ShellCommand(
		name			= "install_database",
		description		= "installing database",
		descriptionDone	= "installed database",
		haltOnFailure	= True,
		command			= ["python", "runInstall.py", "--dba", "--function=install", "--outputDir=./output_install", "--logfile=./output_install/runInstall.log", "--source=.", "--server=" + oracleServer, "--user=" + ORACLE_USERNAME, "--password=" + ORACLE_PASSWORD, "--sysPassword=" + ORACLE_SYS_PASSWORD, "--noStoreLog", "--noFtpUpload", "--client=!!!", "--language=001", "--noConfirm", "--noShowProgress", "--installTest"],
		logfiles		= {"installDatabase": "./output_install/runInstall.log"},
		workdir			= "build/LJS_DB/Install",
		decodeRC		= {
			0: util.SUCCESS,
			1: util.WARNINGS
		}
	))
```

**Comment by doberkofler** [comment]
@p12tic I just double checked the issue and I might be hitting the same bug as described in #4744.

**Comment by p12tic** [comment]
@doberkofler : Could you try to patch your worker code with the patch in this PR #4759? This would make it clear whether the issues are related.

**Comment by doberkofler** [comment]
@p12tic I just replaced the `runprocess.py` file on the worker and the problem is gone, so I guess the issues are related. Would you know when a patch will be released as I have the same problem on all our workers right now?

**Comment by p12tic** [comment]
It will be released as Buildbot 2.4 around in the next week or so.

**Comment by p12tic** [comment]
Closing the issue as it has been fixed in PR #4759. If it reappears, please reopen.
